Contract: Dirty data remains in the contract override

Dirty data remains in the contract override. There is a method to clear the old data but it doesn't seem to always work as intended.

Additionally, objective data was never removed.

I need to confirm if this happens only on a restart or in general per game playthrough (one session)

In this attached log,

```
2020-06-16T13:03:56 FYLS [ERROR] PROBLEM - Can't find [Destroy Capellan Support Lance 1] with id [c0c83bd8-1ab2-40a9-85d2-3227240d00d9] in the encounter layer. Using whatever was configured in the encounterlayer since we can't override it. Expect mechs not to spawn, or very wrong mechs. Who knows, you might get lucky though.
```

Appears in a contract that didn't have (or need) an AL.

[cleaned_output_log.txt](https://github.com/CWolfs/MissionControl/files/4791261/cleaned_output_log.txt)

--- a/src/Core/EncounterFactories/ObjectiveFactory.cs
+++ b/src/Core/EncounterFactories/ObjectiveFactory.cs
@@ -100,7 +100,7 @@ namespace MissionControl.EncounterFactories {
       if (createObjectiveOverride) {
         ObjectiveOverride objectiveOverride = new ObjectiveOverride(destroyLanceObjective);
         objectiveOverride.title = destroyLanceObjective.title;
-        objectiveOverride.description = destroyLanceObjective.description;
+        objectiveOverride.description = "MC" + destroyLanceObjective.description;  // Important and used for objective cleanup
         objectiveOverride.OnSuccessResults = destroyLanceObjective.OnSuccessResults;
         objectiveOverride.OnFailureResults = destroyLanceObjective.OnFailureResults;
         objectiveOverride.OnSuccessDialogueGUID = destroyLanceObjective.onSuccessDialogue.EncounterObjectGuid;

--- a/src/Core/EncounterLogic/ObjectiveLogic/AddFullContractObjective.cs
+++ b/src/Core/EncounterLogic/ObjectiveLogic/AddFullContractObjective.cs
@@ -44,7 +44,7 @@ namespace MissionControl.Logic {
 
       contractObjectiveOverride.isPrimary = isPrimary;
       contractObjectiveOverride.title = title;
-      contractObjectiveOverride.description = description;
+      contractObjectiveOverride.description = "MC" + description;  // Important and used for objective cleanup
       contractObjectiveOverride.objectiveGuids = ObjectiveGuids;
       contractObjectiveOverride.forPlayer = TeamController.Player1;
 

--- a/src/Core/EncounterLogic/ObjectiveLogic/AddPartialContractObjective.cs
+++ b/src/Core/EncounterLogic/ObjectiveLogic/AddPartialContractObjective.cs
@@ -44,7 +44,7 @@ namespace MissionControl.Logic {
 
       contractObjectiveOverride.isPrimary = isPrimary;
       contractObjectiveOverride.title = title;
-      contractObjectiveOverride.description = description;
+      contractObjectiveOverride.description = "MC" + description; // Important and used for objective cleanup
       contractObjectiveOverride.objectiveGuids = ObjectiveGuids;
       contractObjectiveOverride.forPlayer = TeamController.Player1;
 

--- a/src/Core/MissionControl.cs
+++ b/src/Core/MissionControl.cs
@@ -201,10 +201,15 @@ namespace MissionControl {
 
       // Clear old lance data
       if (CurrentContract != null) {
+        // Old Lances
         CurrentContract.Override.targetTeam.lanceOverrideList =
           CurrentContract.Override.targetTeam.lanceOverrideList.Where(lanceOverride => !(lanceOverride is MLanceOverride)).ToList();
         CurrentContract.Override.employerTeam.lanceOverrideList =
           CurrentContract.Override.employerTeam.lanceOverrideList.Where(lanceOverride => !(lanceOverride is MLanceOverride)).ToList();
+
+        // Old Objectives
+        CurrentContract.Override.contractObjectiveList =
+         CurrentContract.Override.contractObjectiveList.Where(contractObjective => !contractObjective.description.StartsWith("MC ")).ToList();
       }
     }
 

